torch/quantization/observer.py

```diff
@@ -232,6 +234,11 @@ def _calculate_qparams(self, min_val, max_val):
                     zero_point = zero_point.new_full(zero_point.size(), (qmin + qmax) // 2)
                 else:
                     zero_point = zero_point.new_full(zero_point.size(), 128)
+        elif self.qscheme == torch.per_channel_affine_float_qparams:
+            scale = (orig_max - orig_min) / float(qmax - qmin)
+            scale_ones = torch.ones_like(scale)
+            scale = torch.where(scale != 0, scale, scale_ones)
```

Comment:
qq, should this be equality comparison or a threshold check against zero?

Reply:
Do you mean check for `scale>0` vs `scale!=0`? In the code we compute inverse scale so we need to set all values of 0 scale to 1.

Reply:
`abs(scale) < eps` instead of `scale != 0`, since scale is a float...?